### PR TITLE
[ENH] Add warning for future change of the default of `force_resample` to True

### PR DESCRIPTION
**Pull request by ajoshiusc**
- closes [2305](https://github.com/nilearn/nilearn/issues/2305).

The force_resample=False which is currently default, the code is not implemented correctly. There are several of #TODO flags in the code for force_resample=False case. Therefore, changed the default option to force_resample=True.

### CODE REVIEW

File: nilearn/image/resampling.py [hunk 1]
```diff
@@ -340,7 +340,7 @@ def resample_img(
     order="F",
     clip=True,
     fill_value=0,
-    force_resample=False,
+    force_resample=True,
```
Comment (bthirion) [inline nilearn/image/resampling.py:343]: Well this is an API change: it has to go through a deprectaion cycle befire we can actually  publish this.

**Comment by Remi-Gau** [comment]
@ajoshiusc I went ahead and added the required changes to implement the deprecation and make the old default value explicit in the tests.

**Comment by Remi-Gau** [comment]
Can you add your contributor information in the [CITATION.CFF](https://github.com/nilearn/nilearn/blob/11a4814492ec5a960f1c27ad9e9649e3a49a56bf/CITATION.cff#L42)

See our contributing guidelines
https://nilearn.github.io/stable/development.html#changelog

authors:

For example

```yml
  - given-names: Dimitri Papadopoulos
    family-names: Orfanos
    website: https://github.com/DimitriPapadopoulos
    affiliation: NeuroSpin, C.E.A., Université Paris-Saclay, Gif-sur-Yvette, France
    orcid: https://orcid.org/0000-0002-1242-8990
```

File: nilearn/image/tests/test_resampling.py
Comment (man-shu) [inline nilearn/image/tests/test_resampling.py:1]: I see that all the tests are for the `force_resample=False` case. 
Would it make sense to have one for the True case?

Comment (Remi-Gau) [reply nilearn/image/tests/test_resampling.py:1]: yeah that makes sense

Comment (bthirion) [reply nilearn/image/tests/test_resampling.py:1]: Please do it.

Comment (Remi-Gau) [reply nilearn/image/tests/test_resampling.py:1]: Added parametrization of all tests so this way we test with the "old" and the "new" default in all tests.

**Comment by man-shu** [comment]
Merged the main into this branch because the `get_data` workflow for the docs builder had failed. Let's see if it passes now.

**Comment by Remi-Gau** [comment]
> Merged the main into this branch because the `get_data` workflow for the docs builder had failed. Let's see if it passes now.

yeah I think this was just a failed request to get the data

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Pull request merged**